Add elementwise_affine for LayerNormGradientOp

xiaomengy opened the pull request:
Summary: Add elementwise_affine for LayerNormGradientOp

Differential Revision: D15157493

houseroad reviewed (approved):
Looks good.

Do we have a performance improvement after fusing bias and scales into LayerNorm(Gradient)?

BIT-silence commented on caffe2/python/helpers/normalization.py:
```diff
@@ -243,20 +243,10 @@ def layer_norm(
     scale = model.create_param(
         param_name='{}_scale'.format(blob_out),
-        shape=[dim_in],
+        shape=[dim_in] if isinstance(dim_in, int) else dim_in,
```
If axis is not the last dimension, dim_in should be the dims from axis to the end.

xiaomengy committed:
Add elementwise_affine for LayerNormGradientOp (#19982)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/19982

Add elementwise_affine for LayerNormGradientOp

Differential Revision: D15157493

fbshipit-source-id: 9049fd4f90e41844f9d796591e4e0cbc6ffb20af
